### PR TITLE
Moved some more dependencies to container

### PR DESCRIPTION
**Pull request by richardmiller-zz**

### CODE REVIEW

File: app/bootstrap.php [hunk 1]
```diff
@@ -63,14 +64,15 @@ function (ConsoleCommandEvent $event) use ($userInteraction) {
 };
 
 $container['installer.docker'] = function ($c) {
+    $dinghyCli = new DinghyCli($c['process.interactive_runner']);
```
Comment (sroze) [inline app/bootstrap.php]: Isn't it more relevant to move it to a container service if it's used in more than one place ?

File: src/Installer/DNS/DockerRouting.php [hunk 2]
```diff
@@ -69,4 +56,58 @@ public function getName()
     {
         return 'routing';
     }
+
+    /**
+     * @param $dinghyIp
```
Comment (sroze) [inline src/Installer/DNS/DockerRouting.php]: Missing a bit of PHPDocs

File: src/Installer/DNS/DockerRouting.php [hunk 3]
```diff
@@ -69,4 +56,58 @@ public function getName()
...
+     * @param $processRunner
+     * @param $userInteraction
+     * @throws \Exception
+     * @throws \Symfony\Component\Process\Exception\ProcessFailedException
```
Comment (sroze) [inline src/Installer/DNS/DockerRouting.php]: Has it `use`d in the file, no need of this FCQN.

File: src/Installer/DNS/DockerRouting.php [hunk 4]
```diff
@@ -69,4 +56,58 @@ public function getName()
     {
         return 'routing';
     }
+
+    /**
+     * @param $dinghyIp
+     * @param $processRunner
+     * @param $userInteraction
+     * @throws \Exception
```
Comment (sroze) [inline src/Installer/DNS/DockerRouting.php]: Is that "true" ?

File: app/bootstrap.php (hunk 1 above)
Comment (richardmiller-zz) [inline app/bootstrap.php]: Will it always be the interactive runner that is used with this?

File: src/Installer/Docker/Dinghy.php [hunk 5]
```diff
@@ -92,4 +82,49 @@ public function getName()
...
+        }
+
+        $this->userInteraction->writeTitle('Installing Dinghy');
+        $this->processRunner->run(
+            new Process('brew install https://github.com/codekitchen/dinghy/raw/latest/dinghy.rb')
+        );
+        $this->userInteraction->writeTitle('Successfully installed Dinghy');
+    }
+
+    private
```
Comment (sroze) [inline src/Installer/Docker/Dinghy.php]: Why ? :)

File: app/bootstrap.php (hunk 1 above)
Comment (richardmiller-zz) [inline app/bootstrap.php]: I guess it doesn't matter, it can just be a different service if it's needed with another process runner

Comment (sroze) [inline app/bootstrap.php]: Yes, I think if a silent process runner is needed, we can just create a new service

File: src/Installer/Docker/Dinghy.php (hunk 5 above)
Comment (richardmiller-zz) [inline src/Installer/Docker/Dinghy.php]: ¯_(ツ)_/¯

Phpstorm reformatting it wierdly I guess :)

**Comment by sroze** [comment]
:+1: :smiley:

**Pull request merged**